jOOQ-kotlin/src/main/kotlin/org/jooq/kotlin/Extensions.kt

```diff
@@ -83,6 +87,15 @@ inline fun Field<Boolean>.not(): Condition = condition(this).not()
 // Extensions to extract fields from Tables
 // ----------------------------------------------------------------------------
 
+@Support
+inline fun <reified T: Any> TableLike<*>.field(index: Int) = this.field(index, T::class.java)
+
+@Support
+inline fun <reified T: Any> TableLike<*>.field(name: Name) = this.field(name, T::class.java)
+
+@Support
+inline fun <reified T: Any> TableLike<*>.field(name: String) = this.field(name, T::class.java)
+
```

Comment:
These types of suggestions have been made before by Jake Wharton
https://github.com/jOOQ/jOOQ/issues/6256#issuecomment-512314479

And later rejected after thorough analysis:
https://github.com/jOOQ/jOOQ/issues/6256#issuecomment-625175287

TL;DR: While looking very enticing at first, it's too late to add such convenience to the vast jOOQ API, as:

- It would be a *huge* change with many many methods involved
- Some methods could not have this kotlin-idiom applied consistently, leading to more cognitive friction than benefit

Reply:
Ah, thanks for the link. These are the only ones I was aware of, but I really don't have a good oversight of the full API. At some point it just becomes too much, I agree.